pkg/registry/apis/dashboard/legacy_storage.go

```diff
@@ -66,19 +66,21 @@ func (s *dashboardStorage) Create(ctx context.Context,
 		return nil, err
 	}
 
+	// Why do we modify p in Save Dashboard?
```

Comment:
Something fishy is happening here, the pointed value gets modified upon save, I wonder if this is intended or a bug.